aws/ecs_cluster.go

```diff
@@ -0,0 +1,150 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+func getAllEcsClustersOlderThan(awsSession *session.Session, region string, excludeAfter time.Time) ([]*string, error) {
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+
+	clusterArns, err := getAllEcsClusters(awsSession)
+	if err != nil {
+		logging.Logger.Errorf("Error getting all ECS clusters")
+		return nil, errors.WithStackTrace(err)
+	}
+
+	var filteredEcsClusters []*string
+	for _, clusterArn := range clusterArns {
+
+		firstSeenTime, err := getFirstSeenEcsClusterTag(awsSession, clusterArn)
+		if err != nil {
+			logging.Logger.Errorf("Error getting the `cloud-nuke-first-seen` tag for ECS cluster with ARN %s", aws.StringValue(clusterArn))
+			return nil, errors.WithStackTrace(err)
+		}
+
+		if firstSeenTime.IsZero() {
+			err := tagEcsClusterWhenFirstSeen(awsSession, clusterArn, time.Now().UTC())
+			if err != nil {
+				logging.Logger.Errorf("Error tagigng the ECS cluster with ARN %s", aws.StringValue(clusterArn))
+				return nil, errors.WithStackTrace(err)
+			}
+		} else {
+			if excludeAfter.After(firstSeenTime) {
```

Comment:
NIT: this could be an `else if`

Reply:
Both versions read fine to me, but very happy to change it 👍

Reply:
No strong preference from me, so your call!